xbmc/network/Network.cpp

```diff
@@ -159,19 +161,35 @@ bool CNetworkBase::IsLocalHost(const std::string& hostname)
   return false;
 }
 
-CNetworkInterface* CNetworkBase::GetFirstConnectedInterface()
+CNetworkInterface* CNetworkBase::GetDefaultInterface()
 {
   CNetworkInterface* fallbackInterface = nullptr;
-  for (CNetworkInterface* iface : GetInterfaceList())
+  if (CServiceBroker::GetSettingsComponent()
+          ->GetAdvancedSettings()
+          ->m_defaultNetworkInterfaceName.empty())
   {
-    if (iface && iface->IsConnected())
+    CLog::Log(LOGDEBUG, "Detecting network interface");
+    for (CNetworkInterface* iface : GetInterfaceList())
     {
-      if (!iface->GetCurrentDefaultGateway().empty())
-        return iface;
-      else if (fallbackInterface == nullptr)
-        fallbackInterface = iface;
+      if (iface && iface->IsConnected())
+      {
+        if (!iface->GetCurrentDefaultGateway().empty())
+          return iface;
+        else if (fallbackInterface == nullptr)
+          fallbackInterface = iface;
+      }
     }
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "Using network interface %s",
```

Comment:
Please update to the current log standards using `{}`.

For more information please see the [libfmt documentation](https://fmt.dev/latest/syntax.html)